packages/spore/src/helper.ts

```diff
@@ -0,0 +1,46 @@
+import { ccc } from "@ckb-ccc/core";
+
+export async function searchOneCellByLock(
+  signer: ccc.Signer,
+): Promise<ccc.Cell | undefined> {
+  let liveCell: ccc.Cell | undefined = undefined;
+  for await (const cell of signer.findCells({
+    scriptLenRange: [0, 1],
+    outputDataLenRange: [0, 1],
+  })) {
+    liveCell = cell;
+    break;
+  }
+  return liveCell;
+}
+
+export async function injectOneCapacityCell(
+  signer: ccc.Signer,
+  tx: ccc.TransactionLike,
+): Promise<ccc.Transaction> {
+  const liveCell = await searchOneCellByLock(signer);
+  if (!liveCell) {
+    const address = await signer.getRecommendedAddress();
+    throw new Error("No live cell found in address: " + address);
+  }
+  let txSkeleton = ccc.Transaction.from(tx);
+  txSkeleton.inputs.push(
+    ccc.CellInput.from({
+      previousOutput: liveCell.outPoint,
+      ...liveCell,
+    }),
+  );
+  txSkeleton.witnesses.push("0x");
```

Comment:
Is it possible to use `Signer.prepareTransaction` after collecting all inputs? Extra witnesses don't look necessary.